src/main/java/codeu/controller/AdminServlet.java

```diff
@@ -149,18 +136,60 @@ void computeAdminStats(HttpServletRequest request) {
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response)
-    throws IOException, ServletException {
+      throws IOException, ServletException {
+
+    if (request.getSession().getAttribute("isAdmin").equals(Boolean.TRUE)) {
```

Comment:
Oh, interesting--the .equals() method keeps you from having to cast?

Also, this will throw a NullPointerException if the "isAdmin" attribute isn't set, right? That's probably fine for the context of this project, but it's something to keep in mind for more robust applications, since "A servlet should be able to handle cases in which the client does not choose to join a session, such as when cookies are intentionally turned off.", according to the HttpSession javadoc: https://javaee.github.io/javaee-spec/javadocs/javax/servlet/http/HttpSession.html

No need to change anything, but this could be part of a bigger "product" discussion of what the experience should be like for non-logged-in users.